### PR TITLE
fix: Add several fixes to the release process

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
- Fix bumping version in bugfix/main branches
- Change image tags in Makefile to released ones when tagging the release

### What issues does this PR fix or reference?
https://github.com/devfile/devworkspace-operator/issues/428
https://github.com/devfile/devworkspace-operator/issues/415

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v7-devworkspaces-operator-e2e, v7-devworkspace-happy-path` to trigger)
    - [ ] `v7-devworkspaces-operator-e2e`: DevWorkspace e2e test
    - [ ] `v7-devworkspace-happy-path`: DevWorkspace e2e test

### CODE REVIEW

File: make-release.sh [hunk 1]
```diff
@@ -109,8 +109,10 @@ fi
 set -e
 
 # change version/version.go file
-sed -i version/version.go -r -e 's#(Version = ")(v?[0-9.]+)(")#\1'"${VERSION}"'\3#g'
-
+sed -i version/version.go -r -e "s#(Version = \")(v?[0-9.]+)(\")#\1${VERSION}\3#g"
+# change image tag in Makefile
+sed -i Makefile -r -e "s#export DWO_IMG \?= quay.io/devfile/devworkspace-controller:.*#export DWO_IMG \?= quay.io/devfile/devworkspace-controller:${VERSION}#g"
```
Comment (sleshchenko) [inline make-release.sh:114]: that works fine but I would say it's double work. Instead maybe move it below and do
```suggestion
sed -i Makefile -r -e "s#export DWO_IMG \?= .*#export DWO_IMG \?= ${DWO_QUAY_IMG}#g"
```
and the same for PROJECT_CLONE

**Review by sleshchenko — COMMENTED** [review]
LGTM and seems to work as far as I tested but I have one suggestion left

File: make-release.sh (hunk 1 above)
Comment (mkuznyetsov) [reply make-release.sh]: ok, thanks, I made an even more generic sed, to just match the image name anywhere in Makefile

**Review by sleshchenko — APPROVED** [review]
Seems to work as expected after I executed changed commands

**Pull request merged**